docs/evm_compatibility/overview.md

```diff
@@ -1,63 +1,155 @@
+The mechanism of gas consumption and calculation of gas fees on Neon EVM differ from Ethereum. Gas fees on Neon EVM are much cheaper than on Ethereum because Solana is the settlement layer. 
 
 > Learn more about the [NEON token and how gas fees work on Neon EVM](../../docs/tokens/gas_fees.md).
 
-Several Solana-specific differences also impact smart contract development.
+### Reentrancy-safe approaches
+
+:::caution
+
+Due to the difference in how gas is calculated in Neon EVM, Solidity's transfer() and send() are NOT reentrancy safe methods.
+:::
+
+Hardcoding gas is one method that programmers may apply to prevent reentrancy attacks. Within the Ethereum/Solidity ecosystem, it is possible to default gas to 2300 gwei by using `transfer()`and `send()`. However, since gas costs are now variable, these methods are [no longer advised](https://consensys.net/diligence/blog/2019/09/stop-using-soliditys-transfer-now/).
+
+### Storage
+
+Several Solana-specific differences also impact smart contract development. In Ethereum EVMs, there are two types of accounts, both of which are associated with a storage map which can be used to read and write arbitrary data:
+
+- Basic accounts store an account balance
+- Code accounts store EVM code
+
+Solana’s Sealevel also provides two accounts which are assigned up to 256 consecutive storage slots; but they have different purposes: 
+
+ - Executable
+
+ > Executable accounts provide immutable storage of executable byte code or an (immutable) proxy address of an account which stores mutable executable byte code.
+
+ - Non-executable
+
+> Since executable accounts are immutable, their application state is stored in non-executable accounts.
+
+The differences go deeper still. In an Ethereum EVM, contracts can only read and write their own storage. In Sealevel, any account’s data can be read or written to by a contract. However, the runtime enforces that only an account’s “owner” is allowed to modify it. Changes by any other programs will be reverted and cause the transaction to fail.
+
+:::info
+
+Learn more in the [Solana wiki](https://solana.wiki/docs/).
+
+:::
+
+### Account mapping
+
+Each Ethereum account involved in a transaction must be mapped to a corresponding Solana account. Any call made to the Ethereum account (e.g. to read balance, execute a transaction, etc.) requires that the Solana account is included to make use of the storage provided.
 
-### Upper limit on number of Accounts
-Neon EVM uses Solana Transaction V0 with an extension via [account lookup tables](https://docs.solana.com/developing/lookup-tables): this limits the maximum number of accounts to 64 in a single transaction. Solana requires that all accounts used in a transaction be specified in order to ensure parallel execution of transactions. 
+<!-- based on item in Slack https://neonlabsworkspace.slack.com/archives/C03CQ8A6WTT/p1683107335962669 >> not sure that this is functional end user support as it stands -->
```

Comment:
@anton-lisanin I did my best to make use of what was in Slack -- not convinced that this fully covers what the end-user may need.

Reply:
Remove entire section? Merge it with "Upper limit on number of accounts"